src/GridParser.ts

```diff
@@ -0,0 +1,63 @@
+import { GridNotFoundErr, headerNotFoundErr, invalidDataTypeErr, invalidValueInColumnErr } from "./constants";
+import { ColumnMetadata, dataTypes, Grid, TableData } from "./types";
+
+export default class GridParser {
+    public parseToTableData(initialDataGrid: Grid): TableData | undefined {
+        if (!initialDataGrid) {
+            return undefined;
+        }
+
+        this.validateGridHeader(initialDataGrid);
+        const data: string[][] = this.parseGridData(initialDataGrid, initialDataGrid.Header);
+        
+        return { columnMetadata : initialDataGrid.Header, data: data };
+    }
+
+    private parseGridData(initialDataGrid: Grid, columnMetadata: ColumnMetadata[]) {
+        const gridData: (string | number | boolean)[][] = initialDataGrid.GridData;
+        if (!gridData) {
+            throw new Error(GridNotFoundErr);
+        }
+        
+        const tableData: string[][] = [];
+        for (const rowData of gridData) {
+            const row: string[] = [];
+            var colIndex: number = 0;
+            for (const prop in rowData) {
```

Comment:
can we be consisted in how we use this?
Either use of as in line 23 or maybe use foreach?